pkg/serviceprovider/github/github.go

```diff
@@ -225,9 +254,13 @@ func (g *Github) CheckRepositoryAccess(ctx context.Context, cl client.Client, ac
 		}
 		ghRepository, _, err := ghClient.Repositories.Get(ctx, owner, repo)
 		if err != nil {
+			var rateLimitError *github.RateLimitError
+			if errors.As(err, &rateLimitError) {
```

Comment:
why not `errors.Is` ?

Reply:
We just want to find out that its type is rateLimitError no?
> As(err error, target interface{}) bool checks if any error in the chain of wrapped errors matches the target. The difference is that this function checks whether the error has a specific type, unlike the Is(), which examines if it is a particular error object.

Reply:
I see, thanks